pyro/contrib/named.py

```diff
@@ -207,3 +207,69 @@ def param(placeholder, *args, **kwargs):
     if not isinstance(placeholder, Object):
         return placeholder  # value was already set
     return placeholder.param(*args, **kwargs)
+
+
+@functools.wraps(pyro.condition)
```

Comment:
I'm inclined to separate these new lifted helpers out into a separate module `pyro.contrib.lifted` or `pyro.contrib.scoped`, since `pyro.contrib.named` can be used independently of these new helpers.